docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md

```diff
@@ -0,0 +1,106 @@
+
+# ERC-20 Transfers and EIP-1559
+
+## Introduction
+In this tutorial we will walk through sending transactions with the tx object and understand how the transaction gets executed, using web3.js version 4x.
+
+## The need for EIP-1559
+ 
+Before Users had to set a single gas price for their transactions. This led to high volatility in gas fees and issues with transactions getting stuck if the gas price wasn't high enough.
+With the introduction of EIP-1559, a minimum fee was introduced which will  automatically adjust based on network congestion
+also a tip for miners was introduced to incentivize them, which will prioritize the transaction.
+
+
+##  Overview
+
+Here is a high-level overview of the steps we will be taking in this tutorial:
+
+1. Setting up  
+2. Get current base fee from the network  
+3. Set max priority fee.  
+4. Construct the transaction object
+5. Sign and send the transaction to the RPC provider
+
+Before we start writing and deploying our contract, we need to set up our environment. For that, we need to install the following:
+
+1. Node.js - Node.js is a JavaScript runtime environment that allows you to run JavaScript on the server-side. You can download it from https://nodejs.org/en/download/  
+
+2. npm - Node Package Manager is used to publish and install packages to and from the public npm registry or a private npm registry. Here is how to install it https://docs.npmjs.com/downloading-and-installing-node-js-and-npm. (Alternatively, you can use yarn instead of npm https://classic.yarnpkg.com/lang/en/docs/getting-started/)
+
+# Getting started
+
+Here we initialize Web3 and set an RPC url inside `const Url;` , add our private key in the `const privateKey;` , then we specify the erc20 token inside the  
+ `const tokenAdress;`
+
+Inside the `async function transferWithEIP1559()`, we initialize a provider, then add account and get the current base fee from the network to store it inside of  
+`const gasPrice;` , next we set the maximum priority fee we are willing to allocate inside  `const maxPriorityFeePerGas;`, to do this we make use of utils package of web3.js, which is imported at the top.
+
+Construct the Transaction object:  
+To send the transaction details to the RPC provider for the transfer, we need to format the details in the tx object(transaction object). The tx object must contain the following:  
+from, to, value, gas, maxPriorityFeePerGas.
+Sign and send the transaction:  
+
+To sign a transaction we perform the `signTransaction` method on the account with the help of the web3.js lib.
+Next we use the `sendSignedTransaction` method with the `signedTx` parameters.
+``` 
+Javascript:
```

Comment:
We use typescript in doc examples. or both TypeScript and JavaScript as tabs like here https://docs.web3js.org/guides/advanced/tree_shaking

Reply:
I'll make this change asap